### PR TITLE
added java solution for problem 39. Combination Sum

### PR DESCRIPTION
**Pull request by whitebeard10**
added java solution for problem 39. Combination Sum

image

### CODE REVIEW

**Comment by md-shamim-ahmad** [comment]
Good works :).
Do you have any plan to add ```39. Combination Sum``` question into ```README.md``` file

File: LeetCode Problems/1 to 100/39. Combination Sum/39. Combination Sum.java [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+        }
+
+        for (int i = start; i < candidates.length; i++) {
+            current.add(candidates[i]);
+            backtrack(result, current, candidates, remaining - candidates[i], i); // Recurse with the same candidate allowed
+            current.remove(current.size() - 1);
+        }
+    }
+
+
```
Comment (md-shamim-ahmad) [inline LeetCode Problems/1 to 100/39. Combination Sum/39. Combination Sum.java:25]: Your solution looks good.
Although these two lines do not affect the correctness but I think we don't need these two lines.
Except for this line, the code follows the ```clean code style``` format.

**Comment by whitebeard10** [comment]
I will be updating readme.md by today, will that be okay?

**Comment by md-shamim-ahmad** [comment]
ok

**Comment by whitebeard10** [comment]
i have changed the ❌ to  ✔️ for the problem no.39, is that ok?

**Comment by md-shamim-ahmad** [comment]
noop this file bro.

look here,
every problem directory have a ```readme``` file where have the original question of leetcode problem
I actually added leetcode question in ```README.md``` file.
So you create a ```README.md``` file into ```39. Combination Sum``` directory which holds the orignal question of the leetcode problem.
You can follow this guideline for [how to add leetcode problem](https://github.com/md-shamim-ahmad/LeetCode-Problem-Solution/blob/main/Contribution%20Guidlines/How%20to%20Add%20LeetCode%20Question.md)

**Comment by whitebeard10** [comment]
thanks for the suggestion, however i see that the readme for problem 39 is already there. So please review and merge this PR, i will be looking to add the readme files for other directories/problems  though. thanks

**Review by md-shamim-ahmad — APPROVED** [review]

**Pull request merged**